src/cpp/client.cpp

```diff
@@ -89,15 +89,15 @@ DataSet Client::get_dataset(const std::string& name)
         tensor_key =
             this->_build_dataset_tensor_key(name, tensor_names[i],
                                             true);
-        Command cmd;
+        GetTensorCommand cmd;
         cmd.add_field("AI.TENSORGET");
         cmd.add_field(tensor_key, true);
         cmd.add_field("META");
         cmd.add_field("BLOB");
         reply = this->_run(cmd);
-        reply_dims = CommandReplyParser::get_tensor_dims(reply);
-        blob = CommandReplyParser::get_tensor_data_blob(reply);
-        type = CommandReplyParser::get_tensor_data_type(reply);
+        reply_dims = cmd.get_tensor_dims(reply);
```

Comment:
If we change to static member function, we won't need to create this ``GetTensorCommand``